packages/core-data/src/utils/test/if-matching-action.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * Internal dependencies
+ */
+import ifMatchingAction from '../if-matching-action';
+
+describe( 'ifMatchingAction', () => {
+	function createEnhancedReducer( predicate ) {
+		const enhanceReducer = ifMatchingAction( predicate );
+		return enhanceReducer( () => 'Called' );
+	}
+
+	it( 'should call reducer if predicate returns true', () => {
+		const reducer = createEnhancedReducer( () => true );
+		const nextState = reducer( undefined, { type: '@@INIT' } );
+
+		expect( nextState ).toBe( 'Called' );
+	} );
+
+	it( 'should not call reducer if predicate returns false', () => {
+		const state = {};
```

Comment:
It would be nice to wrap with `deepFreeze` / `Object.freeze` to ensure there is no mutation.